src/VisualStudio/Core/Def/Snippets/RoslynLSPSnippetParser.cs

```diff
@@ -0,0 +1,250 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.LanguageServices.Extensions;
+
+namespace Microsoft.VisualStudio.LanguageServices.Snippets;
+
+[Export]
+[Shared]
+internal class RoslynLSPSnippetParser
```

Comment:
The internal service does not meet our needs, and the team that created it does not currently have resources to customize it to meet our needs. Since we need the ability to trigger snippet sessions from the completion list for both the semantic snippets and for insert full method call (#12363), we just take a copy that we can use and remove our cross-team dependency.

Reply:
I guess it's still unclear to me why we had to clone this in the first place -- I'd really appreciate it if the PR (or commit messages) gave a bit more background here.